app/models/signon_identity.rb

```diff
@@ -85,7 +97,32 @@ def to_session
       'full_name' => @full_name,
       'user_id' => @user.id,
       'profile_url' => @profile_url,
-      'logout_url' => @logout_url
+      'logout_url' => @logout_url,
+      'available_organisations' => @available_organisations,
+      'current_organisation' => @current_organisation
     }
   end
+
+  def current_estate
+    available_estates.find do |estate|
+      estate.sso_organisation_name == @current_organisation
+    end
+  end
+
+  def available_estates
+    @available_estates ||=
+      begin
+        if @available_organisations.include?(DIGITAL_ORG)
+          Estate.all
```

Comment:
Is there a bug here? By my logic `sso_orgs` will not contain `DIGITAL_ORG` because there's no matching estate, which means that the `available_estates` instance method will never match `if @available_organisations.include?(DIGITAL_ORG)`.

In any case I must say I find this code extremely confusing. I think we can find a clearer way of encapsulating the logic of which orgs are made available.